src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -699,10 +699,14 @@ protected DynamoModel(IStartConfiguration config)
                     // Do nothing for now
                 }
             }
+            //If network traffic is disabled, analytics should also be disabled - this is already done in
+            //our other entry points(CLI,Sandbox etc) - but
+            //not all integrators will use those entry points, some may just create a DynamoModel directly.
+            Analytics.DisableAnalytics = NoNetworkMode || Analytics.DisableAnalytics;
 
             // If user skipped analytics from assembly config, do not try to launch the analytics client
             // or the feature flags client for web traffic reason.
-            if (!IsServiceMode && !areAnalyticsDisabledFromConfig && !Analytics.DisableAnalytics)
+            if (!IsServiceMode && !areAnalyticsDisabledFromConfig && !Analytics.DisableAnalytics && !NoNetworkMode)
```

Comment:
I thought `Analytics.DisableAnalytics = NoNetworkMode || Analytics.DisableAnalytics;` above is already covering it because `!Analytics.DisableAnalytics` is used here?

Reply:
it does cover it, but I think moving forward we really should be using checking `NoNetworkMode` separately everywhere - and I will file a task to look into that - as this bug shows, not everyone uses the same code paths when starting Dynamo if they write their own entry points.